flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/MutationActionList.java

```diff
@@ -0,0 +1,371 @@
+ *  Each {@link MutationAction} can create an HBase {@link Mutation} operation type
+ *  including {@link Put}, {@link Increment}, {@link Append} and {@link Delete}.
+ */
+public class MutationActionList {
+	private final List<MutationAction> actions;
+
+	public MutationActionList() {
+		this.actions = new ArrayList<>();
+	}
+
+	public List<MutationAction> getActions() {
+		return this.actions;
+	}
+
+	/**
+	 * Create a new list of HBase {@link Mutation}s.
+	 *
+	 * @param rowKey row that the created {@link Mutation} list is applied to
+	 * @param writeToWAL enable WAL
+	 * @return a list of HBase {@link Mutation}s
+	 */
+	public List<Mutation> newMutationList(byte[] rowKey, boolean writeToWAL) {
+		List<Mutation> mutations = new ArrayList<>();
+		Put put = null;
+		Increment increment = null;
+		Append append = null;
+		Delete delete = null;
+		boolean rowIsDeleted = false;
+		for (MutationAction action : actions) {
+			switch (action.getType()) {
+				case PUT:
+					if (put == null) {
+						put = new Put(rowKey);
+						mutations.add(put);
+					}
+					if (action.getTs() == -1) {
+						put.addColumn(action.getFamily(), action.getQualifier(), action.getValue());
+					} else {
+						put.addColumn(action.getFamily(), action.getQualifier(), action.getTs(), action.getValue());
+					}
+					break;
+
+				case INCREMENT:
+					if (increment == null) {
+						increment = new Increment(rowKey);
+						mutations.add(increment);
+					}
+					increment.addColumn(action.getFamily(), action.getQualifier(), action.getIncrement());
+					break;
+
+				case APPEND:
+					if (append == null) {
+						append = new Append(rowKey);
+						mutations.add(append);
+					}
+					append.add(action.getFamily(), action.getQualifier(), action.getValue());
+					break;
+
+				// If there are multiple DELETE_ROW actions, only the first one is served
+				case DELETE_ROW:
+					if (!rowIsDeleted) {
+						for (int i = 0; i < mutations.size(); ) {
+							if (mutations.get(i) instanceof Delete) {
+								mutations.remove(i);
+							} else {
+								i++;
+							}
+						}
+						delete = new Delete(rowKey, action.getTs());
+						mutations.add(delete);
+						rowIsDeleted = true;
+					}
+					break;
+
+				case DELETE_FAMILY:
+					if (!rowIsDeleted) {
+						if (delete == null) {
+							delete = new Delete(rowKey);
+							mutations.add(delete);
+						}
+						delete.addFamily(action.getFamily(), action.getTs());
+					}
+					break;
+
+				case DELETE_COLUMNS:
+					if (!rowIsDeleted) {
+						if (delete == null) {
+							delete = new Delete(rowKey);
+							mutations.add(delete);
+						}
+						delete.addColumns(action.getFamily(), action.getQualifier(), action.getTs());
+					}
+					break;
+
+				case DELETE_COLUMN:
+					if (!rowIsDeleted) {
+						if (delete == null) {
+							delete = new Delete(rowKey);
+							mutations.add(delete);
+						}
+						delete.addColumn(action.getFamily(), action.getQualifier(), action.getTs());
```

Comment:
if action.getTs()== -1 then you can call delete.addColumn(byte[] fam, byte[] qual) so that latest version of the specified column can be removed. But for all the above you may have to add so many enums. May be you can check with the Flink community as to how to proceed here. Should we add all the delete related support here or can be done in this JIRA only.